test/compliance/test_kaldi.py

```diff
@@ -202,7 +207,29 @@ def get_output_fn(sound, args):
                 window_type=args[21])
             return output
 
-        self._compliance_test_helper('fbank', 97, 22, get_output_fn)
+        self._compliance_test_helper(self.test_filepath, 'fbank', 97, 22, get_output_fn, atol=1e-3, rtol=1e-1)
+
+    def test_resample_waveform(self):
+        def get_output_fn(sound, args):
+            output = kaldi.resample_waveform(sound, args[1], args[2])
+            return output
+
+        self._compliance_test_helper(self.test_8000_filepath, 'resample', 32, 3, get_output_fn, atol=1e-2, rtol=1e-5)
+
+    def test_resample_waveform_upsample(self):
+        sound, sample_rate = torchaudio.load_wav(self.test_8000_filepath)
+        upsample_sound = kaldi.resample_waveform(sound, sample_rate, sample_rate * 2)
+        self.assertTrue(upsample_sound.size(-1) == sound.size(-1) * 2)
+
+    def test_resample_waveform_downsample(self):
+        sound, sample_rate = torchaudio.load_wav(self.test_8000_filepath)
+        downsample_sound = kaldi.resample_waveform(sound, sample_rate, sample_rate // 2)
+        self.assertTrue(downsample_sound.size(-1) == sound.size(-1) // 2)
+
+    def test_resample_waveform_identity(self):
+        sound, sample_rate = torchaudio.load_wav(self.test_8000_filepath)
+        downsample_sound = kaldi.resample_waveform(sound, sample_rate, sample_rate)
+        self.assertTrue(downsample_sound.size(-1) == sound.size(-1))
 
```

Comment:
Same as line 330 comment. Are we comparing this resample to Kaldi's?

Reply:
We compare them to kaldi's outputs ".ark" files in ``test_resample_waveform`` for the error histogram images above. The other tests are about checking size assumptions

Reply:
Is this something that could be added to the suite of tests?

Reply:
Just to be sure: you added `_test_resample_waveform_accuracy` based on this, right?

Reply:
Yes that is correct 😄